tests/parser_tests.py

```diff
@@ -832,54 +994,72 @@ def test_parse_iso_with_extra_words_at_start_and_end_invalid(self):
+            "Meet me at 2016-05-16T04:05:06.789120 at the restaurant.",
+            "Meet me at 2016-05-16 04:05:06.789120 at the restaurant.",
         ]
 
         for ti in test_inputs:
             with self.assertRaises(ParserError):
                 self.parser.parse_iso(ti)
 
-    def test_parse_iso_with_leading_and_trailing_whitespace(self):
-        self.assertEqual(self.parser.parse_iso("      2016"), datetime(2016, 1, 1))
+    def test_iso8601_basic_format(self):
+        self.assertEqual(self.parser.parse_iso("20180517"), datetime(2018, 5, 17))
 
-        self.assertEqual(self.parser.parse_iso("2016      "), datetime(2016, 1, 1))
+        self.assertEqual(
+            self.parser.parse_iso("20180517T10"), datetime(2018, 5, 17, 10)
+        )
 
         self.assertEqual(
-            self.parser.parse_iso("      2016      "), datetime(2016, 1, 1)
+            self.parser.parse_iso("20180517T105513.843456"),
+            datetime(2018, 5, 17, 10, 55, 13, 843456),
         )
 
         self.assertEqual(
-            self.parser.parse_iso("      2016-05-16 04:05:06.789120      "),
-            datetime(2016, 5, 16, 4, 5, 6, 789120),
+            self.parser.parse_iso("20180517T105513Z"),
+            datetime(2018, 5, 17, 10, 55, 13, tzinfo=tz.tzutc()),
         )
 
         self.assertEqual(
-            self.parser.parse_iso("      2016-05-16T04:05:06.789120      "),
-            datetime(2016, 5, 16, 4, 5, 6, 789120),
+            self.parser.parse_iso("20180517T105513.843456-0700"),
+            datetime(2018, 5, 17, 10, 55, 13, 843456, tzinfo=tz.tzoffset(None, -25200)),
         )
 
-    def test_iso8601_basic_format(self):
-        self.assertEqual(self.parser.parse_iso("20180517"), datetime(2018, 5, 17))
+        self.assertEqual(
+            self.parser.parse_iso("20180517T105513-0700"),
+            datetime(2018, 5, 17, 10, 55, 13, tzinfo=tz.tzoffset(None, -25200)),
+        )
 
         self.assertEqual(
-            self.parser.parse_iso("20180517T10"), datetime(2018, 5, 17, 10)
+            self.parser.parse_iso("20180517T105513-07"),
+            datetime(2018, 5, 17, 10, 55, 13, tzinfo=tz.tzoffset(None, -25200)),
         )
 
+        # mixing formats--this may raise a ParserError in the future
         self.assertEqual(
-            self.parser.parse_iso("20180517T105513.84"),
-            datetime(2018, 5, 17, 10, 55, 13, 840000),
+            self.parser.parse_iso("2018-05-17T105513-0700"),
+            datetime(2018, 5, 17, 10, 55, 13, tzinfo=tz.tzoffset(None, -25200)),
         )
 
         self.assertEqual(
-            self.parser.parse_iso("20180517T105513Z"), datetime(2018, 5, 17, 10, 55, 13)
+            self.parser.parse_iso("20180517T10:55:13-07:00"),
+            datetime(2018, 5, 17, 10, 55, 13, tzinfo=tz.tzoffset(None, -25200)),
         )
 
         self.assertEqual(
-            self.parser.parse_iso("20180517T105513-0700"),
+            self.parser.parse_iso("20180517T105513-07:00"),
```

Comment:
Basic format shouldn't have any colons I think.

Reply:
oh yeah! I included 3 tests for mixing basic and extended formats and noted that it may raise an error in the future. I will just remove the 3 tests as they do not provide value to our suite.